src/ansys/tools/path/path.py

```diff
@@ -233,105 +300,242 @@ def find_ansys(version=None, supported_versions=SUPPORTED_ANSYS_VERSIONS):
     --------
     Within Windows
 
-    >>> from ansys.tools.path import find_ansys
-    >>> find_ansys()
+    >>> from ansys.tools.path import find_mapdl
+    >>> find_mapdl()
     'C:/Program Files/ANSYS Inc/v211/ANSYS/bin/winx64/ansys211.exe', 21.1
 
     Within Linux
 
-    >>> find_ansys()
+    >>> find_mapdl()
     (/usr/ansys_inc/v211/ansys/bin/ansys211, 21.1)
     """
-    versions = _get_available_base_ansys(supported_versions)
-    if not versions:
-        return "", ""
-
-    if not version:
-        version = max(versions.keys())
-
-    elif isinstance(version, float):
-        # Using floats, converting to int.
-        version = int(version * 10)
-
-    try:
-        ans_path = versions[version]
-    except KeyError as e:
-        raise ValueError(
-            f"Version {version} not found. Available versions are {list(versions.keys())}"
-        ) from e
-
-    version = abs(version)
-    if os.name == "nt":
+    ans_path, version = _get_unified_install_base_for_version(version, supported_versions)
+    if is_windows():
         ansys_bin = os.path.join(ans_path, "ansys", "bin", "winx64", f"ansys{version}.exe")
     else:
         ansys_bin = os.path.join(ans_path, "ansys", "bin", f"ansys{version}")
     return ansys_bin, version / 10
 
 
-def is_valid_executable_path(exe_loc):
-    return (
-        os.path.isfile(exe_loc)
-        and re.search(r"ansys\d\d\d", os.path.basename(os.path.normpath(exe_loc))) is not None
-    )
+def _find_installation(product: str, version=None, supported_versions=SUPPORTED_ANSYS_VERSIONS):
+    if product == "mapdl":
+        return find_mapdl(version, supported_versions)
+    raise Exception("unexpected product")
 
 
-def is_common_executable_path(exe_loc):
-    path = os.path.normpath(exe_loc)
-    path = path.split(os.sep)
-    if (
-        re.search(r"v(\d\d\d)", exe_loc) is not None
-        and re.search(r"ansys(\d\d\d)", exe_loc) is not None
-    ):
-        equal_version = (
-            re.search(r"v(\d\d\d)", exe_loc)[1] == re.search(r"ansys(\d\d\d)", exe_loc)[1]
+def find_ansys(version=None, supported_versions=SUPPORTED_ANSYS_VERSIONS):
+    """Obsolete method, use find_mapdl."""
+    return _find_installation("mapdl", version, supported_versions)
+
+
+def is_valid_executable_path(product: str, exe_loc: str):
+    if product == "mapdl":
+        return (
+            os.path.isfile(exe_loc)
+            and re.search(r"ansys\d\d\d", os.path.basename(os.path.normpath(exe_loc))) is not None
         )
+    elif product == "mapdl":
```

Comment:
```suggestion
    elif product == "mechanical":
```
maybe you mean ``mechanical``?